components/tools/OmeroJava/test/integration/ConfigurationServiceTest.java

```diff
@@ -151,4 +159,49 @@ public void testSetConfigValue() throws Exception {
         }
     }
 
+    /**
+     * Tests that the list of supported formats in the DB matches
+     * what is currently supported by BioFormats.
+     * This does not include the <code>FakeReader</code> since it is solely
+     * used for testing.
+     *
+     * @throws Exception
+     *             Thrown if an error occurred.
+     */
+    @Test
+    public void testSupportedFormats() throws Exception {
+        final String ref = "Reader";
+        List<String> toExclude = new ArrayList<String>();
+        toExclude.add("Fake");
+        try (final ImageReader reader = new ImageReader()) {
+            IFormatReader[] readers = reader.getReaders();
+            List<String> values = new ArrayList<String>();
+            for (int i = 0; i < readers.length; i++) {
+                IFormatReader r = readers[i];;
+                String name = r.getClass().getSimpleName();
+                if (name.endsWith(ref)) {
+                    name = name.substring(0, name.length() - ref.length());
+                    if (!toExclude.contains(name)) {
+                        values.add(name);
+                        if (r.hasCompanionFiles()) {
+                            values.add("Companion/"+name);
+                        }
+                    }
+                }
+            }
+            //Load from DB
+            ParametersI param = new ParametersI();
+            String sql = "select f from Format as f";
+            List<IObject> objects = iQuery.findAllByQuery(sql, param);
+            Assert.assertEquals(objects.size(), values.size());
```

Comment:
may not be true: the database can contain additional legacy/weird format values if extant images have them: we do expect that values.size() ≤ objects.size().